Feature/privacy policy-Add validation tests for privacy policy page

adorapham opened the pull request:
Provide the validation tests for the  privacy-policy page on these below items
1-FP logo
2-Privacy Policy Header
3-Existing of each paragraph
4-Correct Email address Links

### What does this pull request aim to achieve? 💯

_Please be concise and link any related issue(s):_

<!--### 🚨Before submitting this pull request🚨:-->

<!--_Please do **NOT** submit this PR if you have not done the following:_-->

- [x] I have checked that no one else is working on similar changes.
- [x] I have read the latest [**README**](README.md) and understand the development workflow.
- [x] The name of this branch is: **`feature/<branch_name>`**, or **`hotfix/<branch_name>`** if this is a hotfix, where `<branch_name>` briefly describes the issue(s) resolved and is prefixed with the issue number(s) (e.g. `1127-1130-update-git-branching-model`). The branch is created in a cloned version of this repo, **not a forked repo**.
- [x] This branch is rebased/merged with the **latest staging** (or the **latest production** if it is a hotfix).
- [x] There are no merge conflicts.
- [x] There are no console warnings and errors.
- [x] On the right hand side of this PR, I have linked the appropriate issue(s) under "Issues" and added a project progress under "Projects".
- [x] The title describes the issue(s) being addressed using format: **`<issue_numbers> - <brief_issue_description>`** (e.g. `#1127/#1130 - Update Git Branching Model`)

adorapham committed:
Add validation tests for privacy policy page
1-FP logo
2-Privacy Policy Header
3-Existing each paragraph
4-Link of Email address

adorapham commented:
Hi All, I create a new pull request with the latest codes basing on the team's feedback. Please review for me. Thank you

jessica-lau commented on cypresse2e/cypress/elements/pages/privacyPolicy.js:
```diff
@@ -0,0 +1,110 @@
+class privacyPolicy {
```
class name should be 'PrivacyPolicy' instead of 'privacyPolicy'

adorapham replied:
Update class name to PrivacyPolicy'

jessica-lau commented on cypresse2e/cypress/elements/pages/privacyPolicy.js:
```diff
@@ -0,0 +1,110 @@
+    return cy.xpath(this.dateDocumented);
+  }
+}
+export default privacyPolicy;
```
class name should be 'PrivacyPolicy' instead of 'privacyPolicy'

adorapham replied:
Done:)

jessica-lau commented on cypresse2e/cypress/integration/e2eTests/privacyPolicy/privacyPolicy.spec.js:
```diff
@@ -0,0 +1,168 @@
+import PrivacyPolicyPage from "../../../elements/pages/privacyPolicy";
```
imported class name should be 'PrivacyPolicy' (same as what's in POM file) instead of 'PrivacyPolicyPage'

adorapham replied:
Done:)

jessica-lau commented on cypresse2e/cypress/integration/e2eTests/privacyPolicy/privacyPolicy.spec.js:
```diff
@@ -0,0 +1,168 @@
+import Logo from "../../../elements/pages/fpLogo";
+
+describe("Private Policy for FightPandemics", () => {
+  const privacyPolicy = new PrivacyPolicyPage();
```
'PrivacyPolicyPage()' should be 'PrivacyPolicy()'

adorapham replied:
Done:)

jessica-lau commented on cypresse2e/cypress/fixtures/privacyPolicyContent.json:
```diff
@@ -0,0 +1,98 @@
+      "email": ""
+    }
+  ]
+}
```
With the IDs that have text containing more than one sentence, maybe consider cutting them down further to one sentence max?

adorapham replied:
I cut down to simple sentence/header of paragraph. since i remove the email, all the paragraph is grouped in one type.  Done :)

jessica-lau commented on cypresse2e/cypress/integration/e2eTests/privacyPolicy/privacyPolicy.spec.js:
```diff
@@ -0,0 +1,168 @@
+      privacyPolicy
+        .getDateDocumented()
+        .invoke("text")
+        .should("eq", "April 23rd, 2020");
```
It might be better if the date is defined as a constant at the top of this spec instead of being hardcoded here, so that it can be easily updated when date changes in the future.

adorapham replied:
Done:)

jessica-lau commented on cypresse2e/cypress/integration/e2eTests/privacyPolicy/privacyPolicy.spec.js:
```diff
@@ -0,0 +1,168 @@
+  function validateEachParaWithOrdList(getMethod, dataType, ParaNum) {
+    const objData = getObjectData(dataType, ParaNum);
+    const text = objData.text;
+    const email = objData.email;
```
A suggestion is to create an email constant for "legal@fightpandemics.com" in this spec file and write a test that will find it throughout the context, instead of adding the email where it already exists in the json file. I'm not sure how often the privacy policy context changes, but the email can be removed or added to different paragraphs at any point. What do you think?

adorapham replied:
@jessica-lau it makes sense thank you :)

adorapham replied:
Done:). I move the email to variable on class object, Json file just contain which paragraph number has email since we have 4 same emails on this page.

jessica-lau reviewed (changes requested):
hey @adorapham please see my comments :)

adorapham committed:
Below are the updated:
-class name change to 'PrivacyPolicy' in privacePolicy.js and privacyPolicy.spec.js
-reduce the sentence to check on each paragraph
-date of document is defined as variable
-remove email information from Json file

adorapham committed:
remove console print

adorapham commented:
@jessica-lau and All, I update with Jessica comment. this time the test goes a lot of faster. thank you for reviewing.

adorapham committed:
Clean up the format

galeza commented on cypresse2e/cypress/integration/e2eTests/privacyPolicy/privacyPolicy.spec.js:
```diff
@@ -0,0 +1,142 @@
+  }
+
+  //to check email each paragraph
+  function checkEmaiEachParagraph(paraNum, email, link) {
```
here small spelling: checkEmai**l**..

adorapham replied:
@galeza done :).  I also update improve the function name and the use of parameter. TY

JayashreeSamantasinghar commented on cypresse2e/cypress/elements/pages/privacyPolicy.js:
```diff
@@ -0,0 +1,107 @@
+      case 18:
+        return cy.xpath(this.eightParaLocWithOrdList);
+      case 19:
+        return cy.xpath(this.ninethParaLocWithOrdList);
```
Hi @adorapham 
Here I can't understand why you took case 110 after case 19, correct me if I am wrong.

adorapham replied:
@JayashreeSamantasinghar ,  since i combine two groups into one group  , 110 mean #10 paragraph and 19  mean #9 paragraph, please see my conversion with Jessica

JayashreeSamantasinghar commented on cypresse2e/cypress/integration/e2eTests/privacyPolicy/privacyPolicy.spec.js:
```diff
@@ -0,0 +1,142 @@
+        var numEmails = privacyPolicyContent.paraPolicyWithEmail.length;
+
+        for (var i = 0; i < numEmails; i++) {
+          checkEmaiEachParagraph(
```
Here I think by mistake you typed checkEmaiEachParagraph instead of "checkEmailEachParagraph"

adorapham replied:
@JayashreeSamantasinghar You seem not viewing the latest source.  I did correct the typo checkEmailEachParagraph

adorapham committed:
Fix the spelling check checkEmailEachParagraph
Make the function name more clear and use the passing parameter

jessica-lau commented:
@adorapham correct me if I'm wrong but I think there's an extra "1" preceding in front of each paragraph id number?

adorapham commented:
> @adorapham correct me if I'm wrong but I think there's an extra "1" preceding in front of each paragraph id number?

@jessica-lau , that is my intention since I combine two types of groups into one type of group to simplifier the coding,  The last  one or two numbers is the order of the paragraph in the page with the numbers.  @jessica-lau do you have any other suggestion? I know it doesn't look good but i don't know a better way, i remove a few of functions

jessica-lau commented:
> > @adorapham correct me if I'm wrong but I think there's an extra "1" preceding in front of each paragraph id number?
> 
> @jessica-lau , that is my intention since I combine two types of groups into one type of group to simplifier the coding, The last one or two numbers is the order of the paragraph in the page with the numbers. @jessica-lau do you have any other suggestion? I know it doesn't look good but i don't know a better way, i remove a few of functions

ahhh okay, I see what you mean. I think it's a little confusing at first look, maybe try something like for the main four paragraphs use "id": "main1", "id": "main2" etc. and then for the sub 15 paragraphs do "id": "sub1", "id": "sub2" etc? This might help clarify.

adorapham commented:
> > > @adorapham correct me if I'm wrong but I think there's an extra "1" preceding in front of each paragraph id number?
> > 
> > 
> > @jessica-lau , that is my intention since I combine two types of groups into one type of group to simplifier the coding, The last one or two numbers is the order of the paragraph in the page with the numbers. @jessica-lau do you have any other suggestion? I know it doesn't look good but i don't know a better way, i remove a few of functions
> 
> ahhh okay, I see what you mean. I think it's a little confusing at first look, maybe try something like for the main four paragraphs use "id": "main1", "id": "main2" etc. and then for the sub 15 paragraphs do "id": "sub1", "id": "sub2" etc? This might help clarify.
@jessica-lau  I think about it too but since they all share the same function   getLocationPara(ParaNumber) to return the location, so that is where i get strange number: ).   BTW, I also put the paraNum in function to try make it a connection 

 it("Get and Validate the PRIVACY POLICY existing for the THIRD paragraph with Order List starting with-**3**. How to exercise your rights", (paraNum = **13**) => {
        validateEachParaExisting(privacyPolicyContent.paraPolicyCx[6], paraNum);
      });
before i have two functions on PrivacyPolicy.js
getLocationPara(ParaNumber) 
getLocationParaWithOrdList(ParaNumber)
please let me know should we move back to two functions?

jessica-lau commented:
> > > > @adorapham correct me if I'm wrong but I think there's an extra "1" preceding in front of each paragraph id number?
> > > 
> > > 
> > > @jessica-lau , that is my intention since I combine two types of groups into one type of group to simplifier the coding, The last one or two numbers is the order of the paragraph in the page with the numbers. @jessica-lau do you have any other suggestion? I know it doesn't look good but i don't know a better way, i remove a few of functions
> > 
> > 
> > ahhh okay, I see what you mean. I think it's a little confusing at first look, maybe try something like for the main four paragraphs use "id": "main1", "id": "main2" etc. and then for the sub 15 paragraphs do "id": "sub1", "id": "sub2" etc? This might help clarify.
> > @jessica-lau  I think about it too but since they all share the same function   getLocationPara(ParaNumber) to return the location, so that is where i get strange number: ).   BTW, I also put the paraNum in function to try make it a connection
> 
> it("Get and Validate the PRIVACY POLICY existing for the THIRD paragraph with Order List starting with-**3**. How to exercise your rights", (paraNum = **13**) => {
> validateEachParaExisting(privacyPolicyContent.paraPolicyCx[6], paraNum);
> });
> before i have two functions on PrivacyPolicy.js
> getLocationPara(ParaNumber)
> getLocationParaWithOrdList(ParaNumber)
> please let me know should we move back to two functions?

@adorapham We can just keep one function and the IDs as they are. I don't think it's too big of a deal. Thanks for making all the other changes : )

adorapham commented:
> > > > > @adorapham correct me if I'm wrong but I think there's an extra "1" preceding in front of each paragraph id number?
> > > > 
> > > > 
> > > > @jessica-lau , that is my intention since I combine two types of groups into one type of group to simplifier the coding, The last one or two numbers is the order of the paragraph in the page with the numbers. @jessica-lau do you have any other suggestion? I know it doesn't look good but i don't know a better way, i remove a few of functions
> > > 
> > > 
> > > ahhh okay, I see what you mean. I think it's a little confusing at first look, maybe try something like for the main four paragraphs use "id": "main1", "id": "main2" etc. and then for the sub 15 paragraphs do "id": "sub1", "id": "sub2" etc? This might help clarify.
> > > @jessica-lau  I think about it too but since they all share the same function   getLocationPara(ParaNumber) to return the location, so that is where i get strange number: ).   BTW, I also put the paraNum in function to try make it a connection
> > 
> > 
> > it("Get and Validate the PRIVACY POLICY existing for the THIRD paragraph with Order List starting with-**3**. How to exercise your rights", (paraNum = **13**) => {
> > validateEachParaExisting(privacyPolicyContent.paraPolicyCx[6], paraNum);
> > });
> > before i have two functions on PrivacyPolicy.js
> > getLocationPara(ParaNumber)
> > getLocationParaWithOrdList(ParaNumber)
> > please let me know should we move back to two functions?
> 
> @adorapham We can just keep one function and the IDs as they are. I don't think it's too big of a deal. Thanks for making all the other changes : )

adorapham commented:
accidently closed :).  I don't know when i close it

idjevm reviewed (changes requested):
Make sure you run prettier or use VSCode to format this file, looks like there are a few things that would get formatted

adorapham committed:
fixing the single quote to double  quote

adorapham commented:
> Make sure you run prettier or use VSCode to format this file, looks like there are a few things that would get formatted

@idjevm, I make a chance for privacyPolicy.js per request, is that all?  btw, when i commit I see the message for all my files ✍️  Fixing up cypresse2e/cypress/elements/pages/privacyPolicy.js.
✅  Everything is awesome!
if you find something else, please let me know. TY

adorapham commented:
**@galeza @jessica-lau @idjevm** , Please check to see if your request commands are fulfill and please let me know if I miss anything.

TimeaKovacs86 reviewed (approved):
Looks Good To Me!

jessica-lau commented:
@adorapham note this will be merged after code freeze on staging is done

galeza reviewed (approved):
LGTM

JayashreeSamantasinghar reviewed (approved):
ran the tests, all passed

galeza commented:
@idjevm 

Can we merge it? Your change's request was corrected?
